app/drivers/sensor/battery/Kconfig

```diff
@@ -6,21 +6,22 @@ DT_COMPAT_ZMK_BATTERY_VOLTAGE_DIVIDER := zmk,battery-voltage-divider
 
 config ZMK_BATTERY
     bool "ZMK battery monitoring"
+    depends on ZMK_BATTERY_REPORTING
     help
         Enable battery monitoring
 
 config ZMK_BATTERY_NRF_VDDH
     bool
     default $(dt_compat_enabled,$(DT_COMPAT_ZMK_BATTERY_NRF_VDDH))
     select ADC
-    select ZMK_BATTERY
+    imply ZMK_BATTERY
```

Comment:
This seems reasonable, but why not also make this `depends on SENSOR` as an alternative?

Reply:
If I understand you correctly that sounds like one of the other alternatives I mentioned:
> Another option to directly disable the symbols would be to add depends on ZMK_BATTERY_REPORTING on each sensor, but this seemed slightly better to me. Let me know if that makes more sense.

But `depends on SENSOR` is probably better that that 😄

Reply:
Although, in that case the battery sensors would still get compiled in if `SENSOR` is enabled through some other symbol like `ZMK_KEYMAP_SENSORS`.

Reply:
I pushed a fixup commit that converts to that; everything builds OK but enabling sensors gets the battery sensors (but not reporting) enabled as well:
```sh
> west build -d build/kl2 -b nice_nano_v2 -- -DSHIELD=kyria_left -DCONFIG_ZMK_BATTERY_REPORTING=n -DCONFIG_EC11=y
[...]
> rg BATTERY build/kl2/zephyr/.config
87:# CONFIG_ZMK_BATTERY_REPORTING is not set
238:CONFIG_DT_HAS_ZMK_BATTERY_NRF_VDDH_ENABLED=y
481:CONFIG_ZMK_BATTERY=y
482:CONFIG_ZMK_BATTERY_NRF_VDDH=y
```

Reply:
I think that's expected behavior, if the nodes are enabled, and the necessary support Kconfig symbols are enabled, we auto-enable the driver. You'd either disable the node in the DTS, or disable it explicitly in the Kconfig as well.

Reply:
Got it, does the current state look good in that case?

Reply:
Yes, I believe this is correct now. Thanks!